windows_integration/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+use tracing::{event, Level};
+use windows::Win32::{
+    Foundation::{NTSTATUS, STATUS_SUCCESS},
+    Security::Authentication::Identity::SECPKG_FUNCTION_TABLE,
+};
+
+mod auth;
+mod security;
+
+/// # Safety
+/// This should only ever be called by the windows api, and FFI with C++ is always unsafe
+/// So beware of demons I guess :shrug:
+#[no_mangle]
+#[allow(non_snake_case)]
+pub unsafe extern "system" fn SpLsaModeInitialize(
+    lsa_version: u32,
+    pkg_ver: *mut u32,
+    pptables: *mut *mut SECPKG_FUNCTION_TABLE,
+    pctables: *mut u32,
+) -> NTSTATUS {
+    event!(Level::INFO, "Initialising kanidm Windows client");
+    event!(
+        Level::INFO,
+        "Local Security Authority Version {}",
+        lsa_version
+    );
+    event!(Level::INFO, "Client Version v{}", env!("CARGO_PKG_VERSION"));
+
+    let package_version_str = format!(
```

Comment:
You may need to change these format strings to pad leading zeros since version 1.0.0 would be greater than 0.10.100 but they would format string to 100 and 10100 respectively.